### PR TITLE
ci: migrate from circleci to github actions

### LINKED ISSUE #194: Migrate from CircleCI to GitHub Actions
This way we can have everything (source, issues, pull requests and CI) in the one location.

### PR DESCRIPTION
**Pull request by codeandcats**
### Migrate from CircleCI to Github Actions
- Remove CircleCI workflow
- Bump required version of Node to 10 since yarn currently complains when trying to use 8 because of commitlint anyway
- Add GitHub Actions workflow
  - Runs build, lint and tests on Node 10, 12, 14 and 15

Closes #194.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+on: push
+
+jobs:
+  check:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [10.x, 12.x, 14.x, 15.x]
```
Comment (laurence-myers) [inline .github/workflows/build.yml]: 🙏 Node 16 was recently released, let's add that too.

Node 10 goes end-of-life and the end up April, I wonder if we should remove support? Maybe we just wait until dependencies start breaking.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v2
+      with:
+        node-version: ${{ matrix.node-version }}
+
+    - name: Get Node.js version
+      id: get-node-version
+      run: echo "::set-output name=node-version::$(node --version)"
+
+    # Next step required for running actions locally via `act`
```
Comment (laurence-myers) [inline .github/workflows/build.yml]: ⛏️ It seems odd to have CI run a step that's only required for testing CI locally.

[Looks like we can make this step conditional](https://github.com/nektos/act#skipping-steps) in `act`:

```yaml
- name: Instal Yarn
  if: ${{ env.ACT }}
  run: npm i yarn -g
```

**Review by laurence-myers — DISMISSED** [review]
Looks good! 🙂 I've left a couple of suggestions.

In the future we can use GitHub Actions to release new versions too.

File: .github/workflows/build.yml (hunk 2 above)
Comment (codeandcats) [reply .github/workflows/build.yml]: Agreed, love it!

**Review by laurence-myers — APPROVED** [review]

**Pull request merged**